File: PostQueryService/PostQueryService.Domain/Post.cs
namespace PostQueryService.Domain
{
    public class Post(
        DateTime createdAt,
        DateTime? updatedAt,
        DateTime? deletedAt,
        bool isDeleted,
        int version,
        Content? content,
        IReadOnlyList<Media.Models.Media> media
    )
    {
        public DateTime CreatedAt { get; private set; } = createdAt;
        public DateTime? UpdatedAt { get; private set; } = updatedAt;
        public DateTime? DeletedAt { get; private set; } = deletedAt;
        public bool IsDeleted { get; private set; } = isDeleted;
        public int Version { get; private set; } = version;
        public Content? Content { get; private set; } = content;
        public IReadOnlyList<Media.Models.Media> Media { get; private set; } = media;
    }
}

File: AngularClient/app_angular_client/src/app/post-read/models/post.ts
import { Media } from "../../media-core/media";
import { PostContent } from "../../post/domain/post-content/post-content";

export interface Post{
  id: string;
  createdAt: Date;
  updatedAt?: Date;
  content?: PostContent;
  media: Array<Media>;
  userId: string;
  name?: string;
  userName: string;
  userMedia?: Media;
}
